astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2082,6 +2097,70 @@ def test_unstructured_to_structured(self):
 
+    def test_merge_arrays(self):
+        """Test `numpy.lib.recfunctions.merge_arrays`."""
+        # merge 1 normal array
+        arr = rfn.merge_arrays(self.q_pv["p"])
+        assert np.array_equal(arr.value["f0"], [1, 2, 3])
+        assert arr.unit == (u.km,)
+
+        # merge 1 structured array
+        arr = rfn.merge_arrays(self.q_pv)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 1-elt tuple
+        arr = rfn.merge_arrays((self.q_pv,))
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 2-elt tuples
+        arr = rfn.merge_arrays((self.q_pv, self.q_pv_t))
+        assert np.array_equal(arr.value["f0"]["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["f0"]["v"], [0.25, 0.5, 0.75])
+        assert np.array_equal(arr.value["f1"]["pv"]["pp"], [4, 5, 6])
+        assert np.array_equal(arr.value["f1"]["pv"]["vv"], [2.5, 5, 7.5])
+        assert np.array_equal(arr.value["f1"]["t"], [0, 1, 2])
+        assert arr.unit == ((u.km, u.km/u.s), ((u.km, u.km/u.s), u.s))
+
+    def test_merge_arrays_flatten(self):
+        """Test `numpy.lib.recfunctions.merge_arrays` with ``flatten=True``."""
+        # merge 1 normal array
+        arr = rfn.merge_arrays(self.q_pv["p"], flatten=True)
+        assert np.array_equal(arr.value["f0"], [1, 2, 3])
+        assert arr.unit == (u.km,)
+
+        # merges 1 array (no need to flatten)
+        arr = rfn.merge_arrays(self.q_pv, flatten=True)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 1-elt tuple (no need to flatten)
+        arr = rfn.merge_arrays((self.q_pv,), flatten=True)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 2-elt tuple
+        arr = rfn.merge_arrays((self.q_pv, self.q_pv_t), flatten=True)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert np.array_equal(arr.value["pp"], [4, 5, 6])
+        assert np.array_equal(arr.value["vv"], [2.5, 5, 7.5])
+        assert np.array_equal(arr.value["t"], [0, 1, 2])
+        assert arr.unit == (u.km, u.km/u.s, u.km, u.km/u.s, u.s)
+
+    def test_merge_arrays_asrecarray(self):
+        with pytest.raises(ValueError, match="asrecarray=True is not supported."):
+            arr = rfn.merge_arrays(self.q_pv, asrecarray=True)
+
+    def test_merge_arrays_usemask(self):
+        with pytest.raises(ValueError, match="usemask=True is not supported."):
+            arr = rfn.merge_arrays(self.q_pv, usemask=True)
+
```

Comment:
Check for failure on a combination of a quantity and a string array?

Reply:
✅